src/main/java/io/jenkins/plugins/report/jtreg/main/comparator/ArgumentsParsing.java

```diff
@@ -0,0 +1,139 @@
+package io.jenkins.plugins.report.jtreg.main.comparator;
+
+import io.jenkins.plugins.report.jtreg.Constants;
+
+public class ArgumentsParsing {
+    // parses the given arguments and returns instance of Options
+    public static Options parse(String[] arguments) {
+        Options options = new Options();
+
+        if (arguments.length >= 1) {
+            for (int i = 0; i < arguments.length; i++) {
+                String[] splitArg = arguments[i].split("=");
+
+                // delete all leading - characters
+                String currentArg = splitArg[0].replaceAll("^-+", "--");
+
+                if (!currentArg.matches("^--.*")) {
+                    throw new RuntimeException("Unknown argument " + arguments[i] + ", did you forget the leading hyphens (--)?");
+                }
+
+                // parsing the arguments:
+                if (currentArg.equals(ArgumentsDeclared.helpArg.getName()) || currentArg.equals("--h")) {
+                    // --help
+                    System.out.print(HelpMessage.HELP_MESSAGE);
+                    System.exit(0);
+
+                } else if (currentArg.equals(ArgumentsDeclared.listArg.getName())) {
+                    // --list
+                    if (options.getOperation() != null) {
+                        throw new RuntimeException("Cannot combine --list with other operations.");
```

Comment:
Majbe this should be handled in the Argument class?

Reply:
I mean this trhow... but maybe it is nonsens. MAybe the argumetn may have some boolean isAlone() or simialrly>